### PR TITLE
fix(side-panel): close with esc key

### LINKED ISSUE #5765: [a11y]: SidePanel should close on `escape` key
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

`2.46.0-rc.0`

### React version

_No response_

### Automated testing tool and ruleset

Manual

### Assistive technology

JAWS

### Description

Since in most cases the `SidePanel` is a dialog, the <kdb>escape</kbd> should close the panel when pressed. The only exception should be for `slideIn` panels, as they are not dialogs and do not trap focus.

### WCAG 2.1 Violation

4.1.2 Name Role Value

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic

### Steps to reproduce

Visit [side panel story](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic), attempt close panel with <kdb>escape</kdb>.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
```[tasklist]
### Tasks
- [ ] Side panel can close with `escape` key, with the exception of `slideIn` side panels
```

### PR DESCRIPTION
**Pull request by ariellalgilmore**
Closes #5765

change side panel to close on `esc` key escept for the `slide in` option
#### What did you change?
hand a `handleEscapeKey` function that both the close icon can use and the entire side panel can use

#### How did you test and verify your work?
storybook

### CODE REVIEW

File: packages/ibm-products/src/components/SidePanel/SidePanel.tsx [hunk 1]
```diff
@@ -728,6 +734,9 @@ export let SidePanel = React.forwardRef(
               className={`${blockClass}__close-button`}
               label={closeIconDescription}
               onClick={onRequestClose}
+              onKeyDown={(event) =>
+                slideIn ? undefined : handleEscapeKey(event)
+              }
               title={closeIconDescription}
```
Comment (amal-k-joy) [inline packages/ibm-products/src/components/SidePanel/SidePanel.tsx:740]: can  we simply this?
`onKeyDown={slideIn ? undefined : handleEscapeKey}`

**Review by amal-k-joy — CHANGES_REQUESTED** [review]
Looking good. One small comment.

File: packages/ibm-products/src/components/SidePanel/SidePanel.tsx (hunk 1 above)
Comment (ariellalgilmore) [reply packages/ibm-products/src/components/SidePanel/SidePanel.tsx:738]: yees thanks! just simplified

**Review by amal-k-joy — APPROVED** [review]

**Review by devadula-nandan — APPROVED** [review]
LGTM

**Pull request merged**